common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -97,20 +97,23 @@ trait Logging {
   }
 
   implicit class LogStringContext(val sc: StringContext) {
-    def log(args: MDC*): MessageWithContext = {
+    def log(args: Any*): MessageWithContext = {
```

Comment:
@gengliangwang I'm not sure if there were discussions on the following case:
```
logInfo(log"MDC(LogKey.ABC, some_string) ${an_object} something")
```

Reply:
In fact, the above case has already been discussed
https://github.com/apache/spark/pull/45813

Reply:
Oh, great, let hold on this one for a while

Reply:
Yeah let's make every variable MDC for now. We can decide which context keys to show in the context by default once we got all the log keys.

Reply:
Changing a log4j configuration will be much easier than debating in logging entries of the code base.

Reply:
> let's make every variable MDC for now

if so, this might not be a good case for the 1st round migration, IMO it's critical to inject a string variable(a large JSON-like pod spec) to the error message

Reply:
@pan3793 ok, we can skip this one